tls/s2n_handshake_io.c

```diff
@@ -1008,6 +1008,7 @@ static int s2n_handshake_write_io(struct s2n_connection *conn)
     POSIX_GUARD(s2n_stuffer_wipe(&conn->handshake.io));
 
     /* Update the secrets, if necessary */
+    POSIX_GUARD_RESULT(s2n_tls13_secrets_update(conn));
```

Comment:
Do these two functions (s2n_tls13_secrets_update and s2n_tls13_key_schedule_update) always need to be called together? Is it better to put them in a single function?

Reply:
Theoretically they're separate operations and could be called separately as long as the secrets are there before the keys need them. In practice, they're called together in the two places they're actually called. It didn't seem worth a whole other function to group them together, especially since that's kind of the idea of s2n_finish_read. You also really need to call s2n_handshake_conn_update_hashes to make the secret derivation work properly.

Calling s2n_tls13_secrets_update directly from s2n_tls13_key_schedule_update would require significantly more updates to the tests due to the setup required for s2n_tls13_secrets_update.

Do you think it's worth another method that calls both?